src/app/api/streaks/route.ts

```diff
@@ -0,0 +1,60 @@
+'use server';
+import { NextResponse } from 'next/server';
+import prisma from '@/lib/db';
+
+//----------------------------------------------------------------
+// this endpoint is used to get all the streaks for a user by id.
+//----------------------------------------------------------------
+export async function GET(req: Request) {
+  try {
+    // get the url from the fetch request
+    const url = new URL(req.url);
+
+    // get the skip parameter to increment the page number
+    const skip = parseInt(url.searchParams.get('skip') || '0', 10);
+
+    const userId = '1';
+    if (!userId) {
+      return NextResponse.json({ error: 'User not found' });
+    }
+    const streaks = await prisma.streaks.findMany({
+      where: {
+        id: userId,
+      },
+      orderBy: {
+        created: 'desc',
+      },
+      take: 5,
+      skip: skip,
+    });
+    return NextResponse.json(streaks);
+  } catch (error) {
+    console.error(error);
+    return NextResponse.json({ error: 'Error getting streaks' });
+  }
+}
+//----------------------------------------------------------------
+//----------------------------------------------------------------
+// Fetch streaks initially and when skip changes
+// useEffect(() => {
+//   const fetchStreaks = async () => {
+//     setIsLoading(true);
+//     try {
+//       const response = await fetch(`/api/streaks?skip=${skip}`);
+//       const newStreaks = await response.json();
+// Append new streaks to the existing list
+//       setStreaks((prev) => [...prev, ...newStreaks]);
+//     } catch (error) {
+//       console.error('Error fetching streaks:', error);
+//     } finally {
+//       setIsLoading(false);
+//     }
+//   };
+
+//   fetchStreaks();
+// }, [skip]);
+
+// Function to load more streaks
+// const handleLoadMore = () => {
+//   setSkip((prevSkip) => prevSkip + 5);
+// };
```

Comment:
I noticed this whole section is commented out, if it's something that we may want to use coming up, then that's probably fine, but otherwise, It's probably best if we remove it, just to make sure our code is clean.

Reply:
yeah its for the future to work with the route.ts